src/services/getResolvedRouteForUrl.browser.spec.ts

```diff
@@ -33,98 +33,93 @@ test('given path WITHOUT params, returns match', () => {
 })
 
 test('given path to unnamed parent, without option to get to leaf, returns undefined', () => {
-  const routes = createRoutes([
-    {
-      path: '/unnamed',
-      children: createRoutes([
-        {
-          name: 'unnamed-child',
-          path: '/unnamed-child/[child-id]',
-          children: createRoutes([
-            {
-              name: 'namedGrandchild',
-              path: '/named-grandchild',
-              component,
-            },
-          ]),
-        },
-      ]),
-    },
-  ])
+  const unnamedParent = createRoute({
+    path: '/unnamed',
+  })
+
+  const unnamedChild = createRoute({
+    parent: unnamedParent,
+    name: 'unnamed-child',
+    path: '/unnamed-child/[child-id]',
+  })
+
+  const namedGrandchild = createRoute({
+    parent: unnamedChild,
+    name: 'namedGrandchild',
+    path: '/named-grandchild',
+    component,
+  })
+
+  const routes = [unnamedParent, unnamedChild, namedGrandchild] as const
 
   const match = getResolvedRouteForUrl(routes, '/unnamed')
 
   expect(match).toBeUndefined()
 })
 
 test('given path to unnamed  parent, with option to get to leaf, returns available leaf', () => {
-  const routes = createRoutes([
-    {
-      path: '/unnamed',
-      children: createRoutes([
-        {
-          name: 'unnamed-child-root',
-          path: '',
-          component,
-        },
-      ]),
-    },
-  ])
+  const unnamedParent = createRoute({
+    path: '/unnamed',
+  })
+
+  const unnamedChildRoot = createRoute({
+    parent: unnamedParent,
+    name: 'unnamed-child-root',
+    path: '',
+    component,
+  })
+
+  const routes = [unnamedChildRoot, unnamedParent] as const
   const match = getResolvedRouteForUrl(routes, '/unnamed')
 
   expect(match?.key).toBe('unnamed-child-root')
 })
 
 test('given path that includes named parent and path to leaf, return first match', () => {
-  const routes = createRoutes([
-    {
-      name: 'namedParent',
-      path: '/named-parent',
-      children: createRoutes([
-        {
-          name: 'namedChild',
-          path: '',
-          children: createRoutes([
-            {
-              name: 'namedGrandchild',
-              path: '',
-              component,
-            },
-          ]),
-        },
-      ]),
-    },
-  ])
+  const namedParent = createRoute({
+    name: 'namedParent',
+    path: '/named-parent',
+  })
 
+  const namedChild = createRoute({
+    parent: namedParent,
+    name: 'namedChild',
+    path: '',
+  })
+
+  const namedGrandchild = createRoute({
+    parent: namedChild,
+    name: 'namedGrandchild',
+    path: '',
```

Comment:
I noticed theres a few places in tests where we're passing an empty path. Which I think we could just omit now that `path` is optional.

Reply:
great call, I removed all instances of `path: ''` from all tests